substrate-node/pallets/pallet-smart-contract/src/tests.rs

```diff
@@ -1177,11 +1213,49 @@ fn test_node_contract_billing_cycles_cancel_contract_during_cycle_works() {
     });
 }
 
+#[test]
+fn test_node_contract_billing_fails() {
+    let (mut ext, mut pool_state) = new_test_ext_with_pool_state(0);
+    ext.execute_with(|| {
+        // Creates a farm and node and sets the price of tft to 0 which raises an error later
+        prepare_farm_and_node_for_failing();
+        run_to_block(1, Some(&mut pool_state));
+
+        assert_ok!(SmartContractModule::create_node_contract(
+            Origin::signed(bob()),
+            1,
+            generate_deployment_hash(),
+            get_deployment_data(),
+            1,
+            None
+        ));
+
+        let contracts_to_bill_at_block = SmartContractModule::contract_to_bill_at_block(11);
+        assert_eq!(contracts_to_bill_at_block.len(), 1);
+
+        // at block 11 billing should fail because of the value of tft being 0
+        // the offchain worker should save the failed ids in local storage and try again
```

Comment:
can we check the offchain worker storage here?

Reply:
Makes sense!